### PR TITLE
Fix scratch layer matching

### PR DESCRIPTION
**Pull request by soraxas**
Error checking when deciding on scratch layer matching (previously there was null exception when `wm_class` was not defined in the user setting. (e.g. when user only want to match by `wm_title`)

I've also commented out the hard-coded gnome-screenshot to scratch, because it keeps on crashing my desktop (see https://github.com/paperwm/PaperWM/issues/376#issuecomment-962700960)

### CODE REVIEW

**Comment by paradajz** [comment]
> I've also commented out the hard-coded gnome-screenshot to scratch, because it keeps on crashing my desktop (see [paperwm#376 (comment)](https://github.com/paperwm/PaperWM/issues/376#issuecomment-962700960))

That's interesting. Screenshot tool was reliably crashing my GNOME as well, but only on next-release branch in official PaperWM repo. Since I switched to next-release branch in this one, it works without crashes. Also some other applications which used to crash the session now work just fine.

**Comment by soraxas** [comment]
I wouldn't say it's reliable (it's a bit random, sometimes it crashes sometimes it doesn't). However, it is still definitely happening to me

File: settings.js [hunk 1]
```diff
@@ -99,11 +99,11 @@ function init() {
...
-    defwinprop({
-        wm_class: /gnome-screenshot/i,
-        scratch_layer: true,
-        focus: true,
-    });
+//    defwinprop({
+//        wm_class: /gnome-screenshot/i,
+//        scratch_layer: true,
+//        focus: true,
+//    });
```
Comment (smichel17) [inline settings.js:106]: Can you move this to a separate commit, and maybe a separate PR? I'm a little lukewarm on this change. According to your comment in https://github.com/paperwm/PaperWM/issues/376#issuecomment-962700960 [1], it doesn't seem to trigger reliably. For myself, I cannot reproduce it with gnome-screenshot… but I *have* infrequently experienced gnome-shell crashing and restarting. So, I suspect this bug is not unique to gnome-screenshot, and we'd be better-served trying to find a root cause than just fixing the symptom on one person's computer.

[1]: Side note: GitHub broke this link for me by appending an `=` to the url when I opened it; in order to *actually* jump to the comment, I had to edit the url to remove the `=` and then hit enter.

**Review by smichel17 — CHANGES_REQUESTED** [review]
As far as I can tell, `wm_title` was never supported (there's no documentation saying it should work, code examples using it, or code implementing it), so it's no surprise that it does not work :slightly_smiling_face:. So, I would consider this a feature request, not a bug fix.

I think it's a reasonable feature request. I could imagine adding custom rules for something like a text editor, which might change its title depending on what file is open (e.g. I want my notes file open in scratch), or browser window, with something like the [Window Titler](https://addons.mozilla.org/en-US/firefox/addon/window-titler/) extension.

Once the question and comment below are addressed, looks good to me :+1:

**Comment by smichel17** [comment]
I rebased on top of upstream's `develop` branch, and updated your branch to remove the comment/uncomment from the commits.

As I mentioned in the other PR, I'd like to move all non-gnome-42 PRs upstream, so I'll close this now; I'd welcome it upstream. Thanks.

**Pull request closed without merging**